decompress.c

```diff
@@ -65,6 +65,68 @@ static sqfs_err sqfs_decompressor_xz(void *in, size_t insz,
 	return SQFS_OK;
 }
 #define CAN_DECOMPRESS_XZ 1
+
+// The following implementation was adapted from  squashfs-tools
+// https://github.com/plougher/squashfs-tools.git
+
+#define LZMA_PROPS_SIZE 5
+#define LZMA_UNCOMP_SIZE 8
+#define LZMA_HEADER_SIZE (LZMA_PROPS_SIZE + LZMA_UNCOMP_SIZE)
+#define MEMLIMIT (32 * 1024 * 1024)
+
+static sqfs_err sqfs_decompressor_lzma(void *in, size_t insz,
+		void *out, size_t *outsz) {
+
+	lzma_stream strm = LZMA_STREAM_INIT;
+	int uncompressed_size = 0, res;
+	unsigned char lzma_header[LZMA_HEADER_SIZE];
+
+	res = lzma_alone_decoder(&strm, MEMLIMIT);
+	if(res != LZMA_OK) {
+		lzma_end(&strm);
+		return SQFS_ERR;
+	}
+
+	memcpy(lzma_header, in, LZMA_HEADER_SIZE);
+	uncompressed_size = lzma_header[LZMA_PROPS_SIZE] |
+		(lzma_header[LZMA_PROPS_SIZE + 1] << 8) |
+		(lzma_header[LZMA_PROPS_SIZE + 2] << 16) |
+		(lzma_header[LZMA_PROPS_SIZE + 3] << 24);
+
+	if(uncompressed_size > *outsz) {
```

Comment:
Please use spaces after `if` to match the rest of the code.

Reply:
done.